src/docker.ts

```diff
@@ -83,7 +83,19 @@ export async function buildImage(impl: string, type: "client" | "server") {
   await new Promise((resolve, reject) => {
     modem.followProgress(
       resp,
-      (err, res) => (err ? reject(err) : resolve(res)),
+      (err, res) => {
+        if (err) {
+          reject(err);
+          return;
+        }
+        for (const entry of res) {
+          if ("error" in entry) {
+            reject(entry.errorDetail.message);
+            return;
+          }
+        }
+        resolve(res);
+      },
```

Comment:
unsure why my machine was getting build failures that were not reporting as true errors :/